### PR TITLE
Feature/add status and type in vulnerabilities table

### LINKED ISSUE #3093: Add the STATUS and TYPE of the vulnerabilities to the vulnerabilities view that affect each agent
**Describe the solution you'd like**

As part of the epic https://github.com/wazuh/wazuh/issues/7749, we are adding new information to the agents' vulnerabilities inventory. This new information will be:

- `status`: It indicates whether the vulnerability is `VALID`, `PENDING`, or `OBSOLETE`.
- `type`: It indicates whether the vulnerability affects the `OS` or a particular `PACKAGE`.

This information should be taken into account to select the vulnerabilities that will be displayed in the vulnerabilities view that affect each agent created as part of the issue https://github.com/wazuh/wazuh-kibana-app/issues/3011.

For simplifying the understanding of the information by the users, we will display only the vulnerabilities in `VALID` and `PENDING` status of any `type` (`OS` or `PACKAGE`). But no information will be provided on the status of the vulnerabilities displayed.

For details about the modifications implemented in the API endpoint used for this, check the issue https://github.com/wazuh/wazuh/issues/7972.

### PR DESCRIPTION
**Pull request by eze9252**
Hi team, this resolve:

- Add fields STATUS and TYPE in vulnerabilities/inventory table
- Refactor in table component (adapting for use in other components)

How to test:

- Changes not ready yet in branch "master" in wazuh api. For test PR, install wazuh from "dev-cves-alerts-inventory" branch.

Captura de pantalla de 2021-04-30 11-21-23

Captura de pantalla de 2021-04-30 11-21-39

Closes #3093

### CODE REVIEW

File: public/components/agents/vuls/inventory/table.tsx [hunk 1]
```diff
@@ -194,27 +163,26 @@ export class InventoryTable extends Component {
...
-        </EuiFlexItem>
-      </EuiFlexGroup>
+        <TableWithSearchBarWzAPI
+          title='Vulnerabilities'
+          tableColumns={columns}
+          tableInitialSortingField='name'
+          searchTable={true}
+          searchBarSuggestions={this.suggestions}
+          endpoint={`/vulnerability/${this.props.agent.id}`}
+          reload={isLoading}
```
Comment (Desvelao) [inline public/components/agents/vuls/inventory/table.tsx:173]: This props in the original table, redo the request with the current table status, when `reload` prop change.

**Comment by Desvelao** [comment]
I was testing the PR with another Wazuh manager branch and I found when exporting to CSV, there columns are missing and the CSV columns are not in the same order that the table in the app.
image

File: public/components/agents/vuls/inventory/table.tsx [hunk 2]
```diff
@@ -194,27 +163,26 @@ export class InventoryTable extends Component {
...
+          isExpandable={true}
+          rowProps={getRowProps}
+          error={error}
+          sorting={sorting}
```
[the remark this answers is not in the document]
Comment (eze9252) [reply public/components/agents/vuls/inventory/table.tsx:177]: I forgot delete this prop.

File: public/components/agents/vuls/inventory/table.tsx [hunk 3]
```diff
@@ -194,27 +163,26 @@ export class InventoryTable extends Component {
...
-            isExpandable={true}
-            loading={isLoading}
-          />
-        </EuiFlexItem>
-      </EuiFlexGroup>
+        <TableWithSearchBarWzAPI
+          title='Vulnerabilities'
+          tableColumns={columns}
+          tableInitialSortingField='name'
+          searchTable={true}
```
[the remark this answers is not in the document]
Comment (eze9252) [reply public/components/agents/vuls/inventory/table.tsx:170]: Yes, I'm renaming the component to `TableWithSearchBardWzAPI`

**Review by frankeros — CHANGES_REQUESTED** [review]
The issue requirements indicate:

>  But no information will be provided on the status of the vulnerabilities displayed.

Remove `Status` and `Type` from columns and detail

**Review by frankeros — APPROVED** [review]

**Pull request merged**